server/services/store/sqlstore/blocks.go

```diff
@@ -346,38 +361,74 @@ func (s *SQLStore) InsertBlock(c store.Container, block model.Block) error {
 			"create_at",
 			"update_at",
 			"delete_at",
-		).Values(
-		c.WorkspaceID,
-		block.ID,
-		block.ParentID,
-		block.RootID,
-		block.ModifiedBy,
-		block.Schema,
-		block.Type,
-		block.Title,
-		fieldsJSON,
-		block.CreateAt,
-		block.UpdateAt,
-		block.DeleteAt,
-	)
-
-	// TODO: migrate this delete/insert to an upsert
-	deleteQuery := s.getQueryBuilder().
-		Delete(s.tablePrefix + "blocks").
-		Where(sq.Eq{"id": block.ID}).
-		Where(sq.Eq{"COALESCE(workspace_id, '0')": c.WorkspaceID})
-	_, err = sq.ExecContextWith(ctx, tx, deleteQuery)
-	if err != nil {
-		_ = tx.Rollback()
-		return err
+		)
+
+	insertQueryValues := map[string]interface{}{
+		"workspace_id":          c.WorkspaceID,
+		"id":                    block.ID,
+		"parent_id":             block.ParentID,
+		"root_id":               block.RootID,
+		s.escapeField("schema"): block.Schema,
+		"type":                  block.Type,
+		"title":                 block.Title,
+		"fields":                fieldsJSON,
+		"delete_at":             block.DeleteAt,
+		"created_by":            block.CreatedBy,
+		"modified_by":           block.ModifiedBy,
+		"create_at":             block.CreateAt,
+		"update_at":             block.UpdateAt,
 	}
 
-	_, err = sq.ExecContextWith(ctx, tx, query.Into(s.tablePrefix+"blocks"))
-	if err != nil {
-		_ = tx.Rollback()
-		return err
+	block.UpdateAt = utils.GetMillis()
+	block.ModifiedBy = userID
+
+	if existingBlock != nil {
+		// block with ID exists, so this is an update operation
+		query := s.getQueryBuilder().Update(s.tablePrefix+"blocks").
+			Where(sq.Eq{"id": block.ID}).
+			Set("workspace_id", c.WorkspaceID).
+			Set("parent_id", block.ParentID).
+			Set("root_id", block.RootID).
+			Set("modified_by", block.ModifiedBy).
+			Set(s.escapeField("schema"), block.Schema).
+			Set("type", block.Type).
+			Set("title", block.Title).
+			Set("fields", fieldsJSON).
+			Set("update_at", block.UpdateAt).
+			Set("delete_at", block.DeleteAt)
+
+		q, args, err := query.ToSql()
+		if err != nil {
+			s.logger.Error("InsertBlock error converting update query object to SQL", mlog.Err(err))
+			return err
+		}
+
+		if _, err := tx.Exec(q, args...); err != nil {
+			s.logger.Error(`InsertBlock error occurred while updating existing block`, mlog.String("blockID", block.ID), mlog.Err(err))
+			return err
+		}
+	} else {
+		block.CreatedBy = userID
+		block.CreateAt = utils.GetMillis()
+		block.ModifiedBy = userID
+		block.UpdateAt = utils.GetMillis()
+
+		insertQueryValues["created_by"] = block.CreatedBy
+		insertQueryValues["create_at"] = block.CreateAt
+		insertQueryValues["update_at"] = block.UpdateAt
+		insertQueryValues["modified_by"] = block.ModifiedBy
+
+		query := insertQuery.SetMap(insertQueryValues)
+		_, err = sq.ExecContextWith(ctx, tx, query.Into(s.tablePrefix+"blocks"))
+		if err != nil {
+			tx.Rollback()
```

Comment:
This error is being ignored.  It should be logged.

Reply:
Done. We weren't logging it anywhere so I've updated it everywhere.